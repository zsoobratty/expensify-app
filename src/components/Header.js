import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink exact to='/' activeClassName='is-active'>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create Expenses</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Help</NavLink>
    </header>
)

export default Header