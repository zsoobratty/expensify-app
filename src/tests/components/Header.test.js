import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Header from '../../components/Header'

test('should render header correctly', () => {
    const wrapper = shallow(<Header />)
    expect(toJSON(wrapper)).toMatchSnapshot()

    // expect(wrapper.find('h1').text()).toBe('Expensify')
    // const renderer = new ReactShallowRender()
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot()

    // console.log(renderer.getRenderOutput())
})