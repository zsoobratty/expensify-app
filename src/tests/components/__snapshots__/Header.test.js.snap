// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <[object Object]
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/create"
  >
    Create Expenses
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/help"
  >
    Help
  </[object Object]>
</header>
`;
